src/gui/qgsfindfilesbypatternwidget.h

```diff
@@ -0,0 +1,98 @@
+/***************************************************************************
+    qgsfindfilesbypatternwidget.h
+    -----------------------------
+    begin                : April 2019
+    copyright            : (C) 2019 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSFINDFILESBYPATTERNWIDGET_H
+#define QGSFINDFILESBYPATTERNWIDGET_H
+
+#include "ui_qgsfindfilesbypatternwidget.h"
+#include "qgis_gui.h"
+
+#include <QDialog>
+
+class QDialogButtonBox;
+
+/**
+ * \class QgsFindFilesByPatternWidget
+ * \ingroup gui
+ * A reusable widget for finding files (recursively) by file pattern.
+ * \since QGIS 3.8
+ */
+class GUI_EXPORT QgsFindFilesByPatternWidget : public QWidget, private Ui::QgsFindFilesByPatternWidgetBase
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsFindFilesByPatternWidget, with the specified \a parent widget.
+     */
+    QgsFindFilesByPatternWidget( QWidget *parent SIP_TRANSFERTHIS = nullptr );
+
+    /**
+     * Returns the list of files found by the dialog.
+     * \see filesFound()
+     */
+    QStringList files() const { return mFiles; }
+
+  signals:
+
+    /**
+     * Emitted after files are found in the dialog.
```

Comment:
Is it worth tweaking the documentation to make it clear that this signal will be emitted even if no files are found (with an empty string list)? 

I'm wondering whether, for clarity, the signal should be renamed to something like searchResult or something.

Reply:
Done, I've used findComplete and updated the dox